src/scenario.jl

```diff
@@ -830,7 +830,7 @@ function cyclone_mortality!(coral_cover, coral_params, cyclone_mortality)::Nothi
 
     # Mid class coral mortality
     coral_mid = hcat(collect(Iterators.partition(coral_params.mid, 4))...)
-    for i in size(coral_mid, 1)
+    for i in 1:size(coral_mid, 1)
```

Comment:
I'm ok with this fix, but maybe a better way would be to do

```
for i in eachrow(coral_mid)
    coral_deaths_mid = coral_cover[i, :] .* cyclone_mortality
    coral_cover[i, :] -= coral_deaths_mid
end
```

Reply:
`eachrow` doesn't produce an index. For that you'd need to use `enumerate`

Reply:
But in this case we don't need an index. The index is only used to access a row of the matrix. I'm proposing replacing:

```
for i in 1:size(coral_mid, 1)
    coral_deaths_mid = coral_cover[coral_mid[i, :], :] .* cyclone_mortality
     coral_cover[coral_mid[i, :], :] -= coral_deaths_mid
end
```` 

by:

```
for i in eachrow(coral_mid)
    coral_deaths_mid = coral_cover[i, :] .* cyclone_mortality
    coral_cover[i, :] -= coral_deaths_mid
end
```

Reply:
`coral_mid` is a matrix of indices so eachrow should work in this case

Reply:
The confusion there is you say you don't need the index, but then use an index?

> for i in eachrow(coral_mid)

> coral_cover[i, :]

Reply:
It makes sense now looking at Dan's changes, thanks.